mir_planning/mir_actions/mir_pick_from_drawer/ros/scripts/pick_from_drawer_server

```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/python
+
+from __future__ import print_function
+import tf
+import copy
+import rospy
+import smach
+
+# import of generic states
+import mir_states.common.manipulation_states as gms
+import mcr_states.common.basic_states as gbs
+import mir_states.common.action_states as gas
+
+# action lib
+from smach_ros import ActionServerWrapper, IntrospectionServer
+
+from std_msgs.msg import String
+from geometry_msgs.msg import PoseStamped, Quaternion
+from mir_planning_msgs.msg import GenericExecuteAction, GenericExecuteResult, GenericExecuteFeedback
+from mir_actions.utils import Utils
+
+#===============================================================================
+
+class SelectObject(smach.State):
+    def __init__(self, topic_name):
+        smach.State.__init__(self,  outcomes=['succeeded'],
+                                    input_keys=['goal'],
+                                    output_keys=['feedback', 'result'])
+        self.publisher = rospy.Publisher(topic_name, String, queue_size=10)
+        rospy.sleep(0.1) # time for the publisher to register in ros network
+
+    def execute(self, userdata):
+        # Add empty result msg (because if none of the state do it, action server gives error)
+        userdata.result = GenericExecuteResult()
+        userdata.feedback = GenericExecuteFeedback(current_state='SelectObject',
+                                                   text='selecting object')
+
+        obj = Utils.get_value_of(userdata.goal.parameters, 'object')
+        self.publisher.publish(String(data=obj))
+        rospy.sleep(0.2) # let the topic survive for some time
+        return 'succeeded'
+        
+#===============================================================================
+
+class SendPoseToMoveIt(smach.State):
+    def __init__(self):
+        smach.State.__init__(self, outcomes=['succeeded', 'failed'])
+        self._moveit_pose_pub = rospy.Publisher('/moveit_to_pose/pose_in',
+                                                PoseStamped, queue_size=10)
+        _pose_sub = rospy.Subscriber('/mcr_perception/object_selector/output/object_pose',
+                                     PoseStamped, self._pose_cb)
+        self._pose = None
+        rospy.sleep(0.1) # time for the publisher to register in ros network
+
+    def _pose_cb(self, msg):
+        self._pose = msg
+
+    def execute(self, userdata):
+        if self._pose is None:
+            return 'failed'
+        arm_target_pose = copy.deepcopy(self._pose)
+        arm_target_pose.header.stamp = rospy.Time.now()
+        arm_target_pose.pose.position.x += 0.0
+        arm_target_pose.pose.position.y += 0.0
+        arm_target_pose.pose.position.z += 0.01
```

Comment:
Fix this when you change from the aruco perception to object perception

Reply:
https://github.com/b-it-bots/mas_industrial_robotics/issues/105